implementations/elixir/ockam/ockam/lib/ockam/transport/tcp/handler.ex

```diff
@@ -28,11 +28,16 @@ defmodule Ockam.Transport.TCP.Handler do
 
     {:ok, socket} = :ranch.handshake(ref, ranch_options)
 
+    # Header, protocol version "1" must be the first thing exchanged.
+    # It isn't send anymore after the initial exchange.
+    {:ok, <<1>>} = transport.recv(socket, 1, 5000)
```

Comment:
I wound change the order of these two lines. Imagine the other side doing the same and now both parties are waiting for the version number.

Reply:
ok, I assumed the connecting side send it, and receiver responded.  But will update